usaspending_api/references/v2/views/filter_tree/psc_filter_tree.py

```diff
@@ -15,18 +15,215 @@
+                    toptier_nodes = self._combine_nodes(toptier_nodes, tier1_nodes)
+                else:
+                    tier1_nodes = self.tier_1_search(tiered_keys, filter_string)
+                    toptier_nodes = self.toptier_search(filter_string, tier1_nodes)
+                toptier_nodes = self._combine_nodes(toptier_nodes, tier1_nodes)
+            else:
+                toptier_nodes = self.toptier_search(filter_string)
+            retval = toptier_nodes
         elif len(tiered_keys) == 1:
-            return self._psc_from_group(tiered_keys[0])
+            if child_layers != 0:
+                tier3_nodes = self.tier_3_search(tiered_keys, filter_string)
+                tier2_nodes = self.tier_2_search(tiered_keys, filter_string, tier3_nodes)
+                tier2_nodes = self._combine_nodes(tier2_nodes, tier3_nodes)
+                tier1_nodes = self.tier_1_search(tiered_keys, filter_string, tier2_nodes)
+                tier1_nodes = self._combine_nodes(tier1_nodes, tier2_nodes)
+            else:
+                tier1_nodes = self.tier_1_search(tiered_keys, filter_string)
+            retval = tier1_nodes
+        elif len(tiered_keys) == 2:
+            if child_layers != 0:
+                tier3_nodes = self.tier_3_search(tiered_keys, filter_string)
+                tier2_nodes = self.tier_2_search(tiered_keys, filter_string, tier3_nodes)
+                tier2_nodes = self._combine_nodes(tier2_nodes, tier3_nodes)
+                retval = tier2_nodes
+            else:
+                if tiered_keys[0] == "Product":
+                    retval = self.tier_3_search(tiered_keys, filter_string)
+                else:
+                    retval = self.tier_2_search(tiered_keys, filter_string)
         else:
-            return self._psc_from_parent(tiered_keys[-1], filter_string)
+            retval = self.tier_3_search(tiered_keys, filter_string)
+        return retval
+
+    def tier_3_search(self, ancestor_array, filter_string) -> list:
+        filters = [Q(length=4)]
+        if ancestor_array:
+            parent = ancestor_array[-1]
+            if len(parent) > 3:
+                filters.append(Q(code__iregex=PSC_GROUPS_COUNT.get(parent, {}).get("pattern") or "(?!)"))
+            else:
+                filters.append(Q(code__startswith=parent))
+        if filter_string:
+            filters.append(Q(Q(code__icontains=filter_string) | Q(description__icontains=filter_string)))
+        retval = []
+        results = PSC.objects.filter(*filters)
+        for object in results:
+            ancestors = []
+            if object.code.isdigit():
+                ancestors.append("Product")
+                ancestors.append(object.code[:2])
+            elif object.code[0] == "A":
+                ancestors.append("Research and Development")
+                ancestors.append(object.code[:2])
+                if object.code[:2] != "AU":
```

Comment:
Is this a new business rule?

Reply:
yeah, `AU` codes skip the length=3 codes and go straight to length=4